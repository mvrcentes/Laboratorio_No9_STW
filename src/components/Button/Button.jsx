import style from "./Button.module.css"

export const Button = ({ name, value, setSelected, backgroundColor }) => {
    return (
        <div
            className={`${style.Button} ${style[`Button__${backgroundColor}`]}`}
            value={value}
            onClick={() => setSelected(value)}
            >
            {name}
        </div>
    )
}
