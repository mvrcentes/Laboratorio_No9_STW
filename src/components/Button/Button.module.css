.Button {
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 25px;
    height: 100px;
    width: 100px;
    background-color: #444444;
    color: #fff;
    font-size: 2rem;
}

.Button:hover {
    background-color: #333333;
    cursor: pointer;
}

.Button__operator {
    background-color: #942aea;
    color: #c086e7;
    opacity: 0.6;
}

.Button__operator:hover {
    background-color: #942aea;
    opacity: 0.3;
}

.Button__equal {
    height: 215px;
    background-color: #942aea;
    color: #fff;
}

.Button__equal:hover {
    background-color: #942aea;
    opacity: 0.6;
}

.Button__AC {
    background-color: #fcb82c;
    color: #fff;
}

.Button__AC:hover {
    background-color: #fcb82c;
    opacity: 0.6;
}