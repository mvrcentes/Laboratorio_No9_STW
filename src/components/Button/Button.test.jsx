import React from "react"
import { render, screen } from "@testing-library/react"


import { Button } from "./Button"

describe("Button", () => {
    it("should render correctly", () => {
        const { container } = render(<Button />)
        expect(container).toMatchSnapshot()
    })

    // Tests that the button component renders with all provided props.
    it("test_button_renders_with_all_props", () => {
        render(
            <Button
                name="Test Button"
                value="test"
                setSelected={() => {}}
                backgroundColor="blue"
            />
        )
        expect(screen.getByText("Test Button")).toBeInTheDocument()
        expect(screen.getByText("Test Button").getAttribute("value")).toBe(
            "test"
        )
    })

    it("test_on_click_invokes_set_selected", () => {
        const setSelectedMock = jest.fn()
        const wrapper = render(
            <Button
                name="Test Button"
                value="test"
                setSelected={setSelectedMock}
                backgroundColor="blue"
            />
        )
        wrapper.simulate("click")
        expect(setSelectedMock).toHaveBeenCalledWith("test")
    })    
})
