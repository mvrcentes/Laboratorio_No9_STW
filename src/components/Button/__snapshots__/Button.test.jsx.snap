// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should render correctly 1`] = `
<div>
  <div
    class="_Button_2a1efe _Button__undefined_2a1efe"
  />
</div>
`;
